### ISSUE TITLE
Potential Windows incompatibility due to system() command

### ISSUE THREAD

**Issue by dmcglinn** [ropensci/ecoretriever#32]
On package install 

`system('retriever update')`

fails with the following error message

```
Cannot open c:\Anaconda\Scripts\RETRIE~1-script.py
Warning message:
running command 'retriever update' had status 2
```

I am not sure yet if this bug generally applies to Windows or is cropping up because of a particular system configuration. The `ecoretriever` package is failing on a Windows 7, 64-bit install running R v3.1.1 64 bit due to the fact that the function `system` which is used to execute `retriever` commands at the CLI is not able to access the `retriever` commands for some reason. The `system` function is executing other windows commands such as `ls`, and the R function `shell` which is a windows specific function is properly executing `retriever` commands. I need to document if this is a general windows bug or just something wonky with my setup.

**Comment by ethanwhite** [comment]
Are you still having issues with this @dmcglinn?

**Comment by dmcglinn** [comment]
It's working now. I think it was having trouble because `system` was trying to access retriever.exe from the `./Anaconda/scripts/` path. Uninstalling anaconda and then reinstalling the retriever fixed the problem. I think this can be closed.

**Comment by ethanwhite** [comment]
Thanks.

**Comment by ethanwhite** [comment]
I'm now seeing this issue on a Windows with no anaconda installation and using a pip installed version of the retriever.

**Comment by ethanwhite** [comment]
@karthik - are there any R-Windows experts on the ropensci team that might be able to consult with us on this?

**Comment by karthik** [comment]
Hi Ethan, absolutely. @jeroenooms should be able to help here.

**Comment by ethanwhite** [comment]
Thanks @karthik!

@jeroenooms - in short, the ecoretriever is an R wrapper of the command line interface for https://github.com/weecology/retriever (a Python program). We currently interact with the retriever CLI using `system()` calls. On the Windows systems we developed/tested on this worked fine, but it appears that on some Windows systems (including 2 out of 3 in my lab right now) `system()` fails when running retriever commands and returns errors like

```
> system('retriever ls')
Cannot open c:\ProgramFiles\Python27\...\RETRIE~1-script.py
Warning message:
running command 'retriever update' had status 2
```

On the systems where `system()` doesn't work, `shell()` does work.

So, basically we're trying to understand why `system` works on some Windows systems but not others, and whether we can get it to work consistently or if we need to check the operating system and use `shell` when running on Windows.

Also looping in @henrykironde the software engineer in my group.

**Comment by jeroenooms** [comment]
Did you try using `system2` instead of `system` ? I think that one is more portable.

**Comment by ethanwhite** [comment]
I did try `system2` on the systems that are currently failing and got the same error. Sorry for not noting that originally.

**Comment by henrykironde** [comment]
```
> system('retriever ls')
Cannot open c:\ProgramFiles\Python27\...\RETRIE~1-script.py
Warning message:
running command 'retriever update' had status 2
```
Based on the output,  How about if we try to specify the full path
```
path <- "PATH TO RETRIVER"
system2(path, "ls")
```

**Comment by jeroenooms** [comment]
Also use `normalizePath(path, mustWork = TRUE)` to make sure you get the full path in the format required for the platform.

**Comment by ethanwhite** [comment]
That didn't fix anything unfortunately.

```
> system2(normalizePath("/Python27/Scripts/retriever.exe"), "ls")
Cannot open C:\Python27\Scripts\RETRIE~1-script.py
Warning message:
running command '"C:\Python27\Scripts\retriever.exe" ls' had status 2
```

**Comment by jeroenooms** [comment]
Can you add the `mustWork=TRUE`? That won't fix the problem but it will tell you if the path is correct.

**Comment by ethanwhite** [comment]
Oops, sorry about that.

```
> system2(normalizePath("/Python27/Scripts/retriever.exe", mustWork = TRUE), "ls")
Cannot open C:\Python27\Scripts\RETRIE~1-script.py
Warning message:
running command '"C:\Python27\Scripts\retriever.exe" ls' had status 2
```

**Comment by jeroenooms** [comment]
Wait so you're calling an executable `retriever.exe` which tries to run a python script? Is that intended? It looks then like the bug is in your `retriever.exe` program which fails to invoke python properly?

**Comment by ethanwhite** [comment]
Yes, it is intended. `retriever.exe` is automatically created by Python's system for producing command line interfaces. It's generated just by installing the module in the standard way. It works totally fine from the command line. Apparently on Windows this is all handled by Python creating an  `exe` that ends up calling an associated Python script.

In contrast to running it from the command line, running `retriever.exe` using  `system()` or `system2()` produces the above error. Running it from `shell()` works fine.

Thanks again for looking into this.

**Comment by ethanwhite** [comment]
I have what seems like a bit of a hack working in #67. I'd welcome thoughts, concerns about the approach, or more satisfying alternatives.
